packages/material-ui/src/styles/createTypography.js

```diff
@@ -35,13 +34,6 @@ export default function createTypography(palette, typography) {
     ...other
   } = typeof typography === 'function' ? typography(palette) : typography;
 
-  warning(
```

Comment:
@oliviertassinari The issue is that `theme.typography` is public so people might use this without touching the `Typography` component (which we also do). We can't really track read-access to deprecated variants. It's not perfect at the moment but it should cover 90% of the use cases.